### PR TITLE
[10.x] Fix file race condition after view:cache and artisan up

### PR DESCRIPTION
**Pull request by roxik**
This PR will fix error with file race condition after call `php artisan view:cache` and `php artisan up`.

Case `php artisan up`:

User has request, passed condition `if ($this->app->maintenance Mode()->active())` then file framework/down deleted by command and then in `file_get_contents` we have `ErrorException`.

Solution: recheck file exists

image

Case `php artisan view:cache`:

Same conditions, but instead `file_get_contents` we have `filemtime`

### CODE REVIEW

**Comment by taylorotwell** [comment]
Please remove all textual formatting / spacing changes made to docblocks and mark as ready for review again when complete. Thanks.

File: src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php [hunk 1]
```diff
@@ -42,6 +43,7 @@ public function __construct(Application $app)
      * @return mixed
      *
      * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+     * @throws ErrorException
```
Comment (driesvints) [inline src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php:46]:
```suggestion
     * @throws \ErrorException
```

File: src/Illuminate/View/Compilers/Compiler.php [hunk 2]
```diff
@@ -89,6 +90,8 @@ public function getCompiledPath($path)
      *
      * @param  string  $path
      * @return bool
+     *
+     * @throws ErrorException
```
Comment (driesvints) [inline src/Illuminate/View/Compilers/Compiler.php:94]:
```suggestion
     * @throws \ErrorException
```

**Review by driesvints — CHANGES_REQUESTED** [review]

**Comment by roxik** [comment]
@driesvints resolved

**Pull request merged**